Lib/unittest/test/testmock/testpatch.py

```diff
@@ -1660,20 +1661,19 @@ def test_mock_calls_with_patch(self):
 
 
     def test_patch_imports_lazily(self):
-        sys.modules.pop('squizz', None)
-
         p1 = patch('squizz.squozz')
         self.assertRaises(ImportError, p1.start)
 
-        squizz = Mock()
-        squizz.squozz = 6
-        sys.modules['squizz'] = squizz
-        p1 = patch('squizz.squozz')
-        squizz.squozz = 3
-        p1.start()
-        p1.stop()
-        self.assertEqual(squizz.squozz, 3)
+        with swap_attr(sys, 'modules', sys.modules.copy()):
```

Comment:
I have doubts about changing the reference to `sys.modules`. it can be cached in some places. It is safer to change `sys.modules` in-place. See for example `sys_modules_context` in `Lib/test/test_importlib/test_namespace_pkgs.py`. There are several other helpers in modules test.support and test.test_importlib.util.

Reply:
Okay with your suggestion, but could you please explain to me what kind of risk is posed by a change in reference?

I like the `sys_modules_context` helper more. Where do you think I should move this function to, before using it in the tests?

Reply:
I don't know if @brettcannon or any of the other importlib maintainers expect the tests to be standalone, but if not then you could move `sys_modules_context` into `test.support`. Alternatively, it could be copied into `test.support` and leave the importlib tests using their own copy.

Reply:
> Okay with your suggestion, but could you please explain to me what kind of risk is posed by a change in reference?

I don't remember details (and these details were changed in recent releases), but the comment to `sys_modules_context` implies that there is such risk.

> Where do you think I should move this function to, before using it in the tests?

You can just import it.
```python
from test.test_importlib.test_namespace_pkgs import sys_modules_context
```
There are many import-related helpers scattered through different files, and I think that once we should merge and standardize them, but this is a different issue.

Reply:
Thanks. I have made the modifications.